src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -21,255 +169,1403 @@ const ActorInit En_Nb_InitVars = {
+        player->actor.shape.rot.y = temp;
+        player->actor.posRot.rot.y = temp;
+    }
+}
+
+void EnNb_SetupChamberWarpImpl(EnNb* this, GlobalContext* globalCtx) {
+    CutsceneContext* csCtx = &globalCtx->csCtx;
+    CsCmdActorAction* csCmdNPCAction;
+
+    if (csCtx->state != 0) {
+        csCmdNPCAction = csCtx->npcActions[1];
+        if (csCmdNPCAction != NULL && csCmdNPCAction->action == 2) {
+            this->action = NB_CHAMBER_APPEAR;
+            this->drawMode = NB_DRAW_DEFAULT;
+            EnNb_SpawnBlueWarp(this, globalCtx);
+        }
+    }
+}
+
+void EnNb_SetupDefaultChamberIdle(EnNb* this) {
+    if (this->actor.shape.unk_08 >= 0.0f) {
+        this->action = NB_CHAMBER_IDLE;
+        this->actor.shape.unk_08 = 0.0f;
+    }
+}
+
+void EnNb_SetupArmRaise(EnNb* this, GlobalContext* globalCtx) {
+    AnimationHeader* animation = &D_0600274C;
+    CsCmdActorAction* csCmdNPCAction;
+
+    if (globalCtx->csCtx.state != 0) {
+        csCmdNPCAction = globalCtx->csCtx.npcActions[1];
+        if (csCmdNPCAction != NULL && csCmdNPCAction->action == 3) {
+            SkelAnime_ChangeAnim(&this->skelAnime, animation, 1.0f, 0.0f,
+                                 SkelAnime_GetFrameCount(&animation->genericHeader), 2, 0.0f);
+            this->action = NB_CHAMBER_RAISE_ARM;
+        }
+    }
+}
+
+void EnNb_SetupRaisedArmTransition(EnNb* this, UNK_TYPE arg1) {
+    AnimationHeader* animation = &D_06002B4C;
+
+    if (arg1 != 0) {
+        SkelAnime_ChangeAnim(&this->skelAnime, animation, 1.0f, 0.0f,
+                             SkelAnime_GetFrameCount(&animation->genericHeader), 0, 0.0f);
+        this->action = NB_CHAMBER_RAISE_ARM_TRANSITION;
+    }
+}
+
+void EnNb_SetupMedallion(EnNb* this, GlobalContext* globalCtx) {
+    CsCmdActorAction* csCmdNPCAction;
+
+    if (globalCtx->csCtx.state != 0) {
+        csCmdNPCAction = globalCtx->csCtx.npcActions[6];
+        if (csCmdNPCAction != NULL && csCmdNPCAction->action == 2) {
+            this->action = NB_GIVE_MEDALLION;
+            EnNb_GiveMedallion(this, globalCtx);
+        }
+    }
+}
+
+// Action func is never explicitly set to this, but it runs when the memory gets zero cleared
+void EnNb_SetupChamberCs(EnNb* this, GlobalContext* globalCtx) {
+    EnNb_SetupChamberCsImpl(this, globalCtx);
+}
+
+void EnNb_SetupChamberWarp(EnNb* this, GlobalContext* globalCtx) {
+    EnNb_SetupChamberWarpImpl(this, globalCtx);
+}
+
+void EnNb_ComeUp(EnNb* this, GlobalContext* globalCtx) {
+    EnNb_ComeUpImpl(this, globalCtx);
+    EnNb_FrameUpdateMatrix(this);
+    EnNb_UpdateEyes(this);
+    EnNb_SetupDefaultChamberIdle(this);
+}
+
+void func_80AB193C(EnNb* this, GlobalContext* globalCtx) {
+    func_80AB1284(this, globalCtx);
+    EnNb_FrameUpdateMatrix(this);
+    EnNb_UpdateEyes(this);
+    EnNb_SetupArmRaise(this, globalCtx);
+}
+
+void EnNb_RaiseArm(EnNb* this, GlobalContext* globalCtx) {
+    s32 unk;
+
+    func_80AB1284(this, globalCtx);
+    unk = EnNb_FrameUpdateMatrix(this);
```

Comment:
this unk comes from `skelAnime->animUpdate(skelAnime)`
in my experience animUpdate functions return a boolean indicating if the animation is finished (for example always 0 if the mode says to loop)
that may make sense given its usage in `EnNb_SetupRaisedArmTransition`?